Minio.Tests/EndpointTest.cs

```diff
@@ -0,0 +1,190 @@
+﻿/*
+ * Minio .NET Library for Amazon S3 Compatible Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+using System;
+using System.Text;
+using System.Collections.Generic;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Net;
+using System.Configuration;
+using Minio.Exceptions;
+using Minio;
+using Minio.Helper;
+namespace Minio.Tests
+{
+    [TestClass]
+    public class EndpointTest
```

Comment:
All validation happens in the umbrella function ValidateEndPoint that combines both URL validity and endpoint checks. GetEndpointURL already checks for domain validity and IP address, so the Inetaddress validation in java is not repeated inside the isValidEndpoint private function.